packages/main/src/RangeSlider.js

```diff
@@ -401,27 +618,71 @@ class RangeSlider extends SliderBase {
 	}
 
 	/**
-	 * Swaps start and end values and handles (thumbs), if one came accros the other
+	 * Swaps the start and end values of the handles if one came accros the other:
+	 * - If the start value is greater than the endValue swap them and their handles
+	 * - If the endValue become less than the start value swap them and their handles
+	 *
+	 * Switches the focus to the opposite of the currently focused handle.
+	 *
+	 * Note: Only the property values are reversed, the DOM elements of the handles
+	 * corresponding to them are never switched.
 	 *
 	 * @private
 	 */
 	_swapValues() {
-		// If the start value is greater than the endValue swap them and their handles
-		if (this._valueAffected === "startValue" && this.startValue > this.endValue) {
-			const oldEndValue = this.endValue;
+		const affectedValue = this._getAffectedValue();
+
+		if (affectedValue === "startValue" && this.startValue > this.endValue) {
+			const prevEndValue = this.endValue;
 			this.endValue = this.startValue;
-			this.startValue = oldEndValue;
-			return;
+			this.startValue = prevEndValue;
+
+			this._setValuesAreReversed();
+			this.focusInnerElement();
 		}
 
-		// If the endValue become less than the start value swap them and their handles
-		if (this._valueAffected === "endValue" && this.endValue < this.startValue) {
-			const oldStartValue = this.startValue;
+		if (affectedValue === "endValue" && this.endValue < this.startValue) {
+			const prevStartValue = this.startValue;
 			this.startValue = this.endValue;
-			this.endValue = oldStartValue;
+			this.endValue = prevStartValue;
+
+			this._setValuesAreReversed();
+			this.focusInnerElement();
 		}
 	}
 
+	/**
+	 * Flag that we have swapped the values of the 'start' and 'end' properties,
+	 * to correctly switch the focus within the component from one handle to another
+	 * when the swapping is finished. As we only swap property values and not
+	 * the handle elements themselves, we must also swap their focus.
+	 *
+	 * @private
+	 */
+	_setValuesAreReversed() {
+		this._reversedValues = !this._reversedValues;
+	 }
+
+	 _areValuesReversed() {
+		return this._reversedValues;
+	}
+
+	get _startHandle() {
+		return this.shadowRoot.querySelector(".ui5-slider-handle--start");
+	}
+
+	get _endHandle() {
+		return this.shadowRoot.querySelector(".ui5-slider-handle--end");
+	}
+
+	get _progressBar() {
+		return this.shadowRoot.querySelector(".ui5-slider-progress");
+	}
+
+	get tabIndexProgress() {
+		return this.tabIndex;
```

Comment:
In the upcoming accessibility commit implementing the aria attributes according to the standards we are removing both this getter and the tabindex private property because they will be be no longer needed in results of a couple template changes of the Slider and & Range Slider components.